src/file_name_manager.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2023 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+
+#include "include/flatbuffers/file_manager.h"
+
+namespace flatbuffers {
+
+class FileNameManager ::FileManager {
```

Comment:
not aware that this style would work. Isn't it `class FileNameSavingFileManager : public FileManager {`?